### PR TITLE
Add `L.VGG19Layers`

### PR DESCRIPTION
**Pull request by crcrpar**
access #5209 

I couldn't run the test with GPUs due to the lack of computational resources.

### CODE REVIEW

**Comment by crcrpar** [comment]
I have one question: Where to locate VGG19 Layers?

Now I locate VGG19Layers in a different file from VGG16Layers because `VGG16Layers` is hard coded in `_make_npz` function and I didn't want to change the existent file.

**Comment by kmaehashi** [comment]
Thanks for the PR!

> I have one question: Where to locate VGG19 Layers?

I think it's better to put VGG19 into `vgg.py` for consistency, 

> Now I locate VGG19Layers in a different file from VGG16Layers because VGG16Layers is hard coded in _make_npz function and I didn't want to change the existent file.

As `_make_npz` is an internal interface used by VGG16Layers, you can change it :)
It seems also better to define `VGGLayers` and let `VGG16Layers` / `VGG19Layers` inherit from it.

**Comment by crcrpar** [comment]
Thank you for your reply and I'll give it a try following your comment.

File: chainer/links/model/vision/vgg.py [hunk 1]
```diff
@@ -66,26 +66,33 @@ class VGG16Layers(link.Chain):
...
+            either 16 or 19.
 
     Attributes:
         available_layers (list of str): The list of available layer names
             used by ``forward`` and ``extract`` methods.
 
     """
 
-    def __init__(self, pretrained_model='auto'):
+    def __init__(self, pretrained_model, n_layers):
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: In `__init__`,
* Why should the default value of argument `pretrained_model='auto'` be removed?
* The change/removal of comment seems unnecessary.

File: chainer/links/model/vision/vgg.py [hunk 2]
```diff
@@ -168,18 +164,19 @@ def forward(self, x, layers=None, **kwargs):
         .. warning::
 
            ``test`` argument is not supported anymore since v2.
-           Instead, use ``chainer.using_config('train', train)``.
+           Instead, use ``chainer.using_config('train', False)``.
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: For clarity, could you add `to run in test mode` at the end of the sentence?

File: chainer/links/model/vision/vgg.py [hunk 3]
```diff
@@ -168,18 +164,19 @@ def forward(self, x, layers=None, **kwargs):
...
             x (~chainer.Variable): Input variable. It should be prepared by
                 ``prepare`` function.
             layers (list of str): The list of layer names you want to extract.
+                If ``None``, 'prob' will be used as layers.
 
         Returns:
-            Dictionary of ~chainer.Variable: A directory in which
-            the key contains the layer name and the value contains
-            the corresponding feature map variable.
+            Dictionary of ~chainer.Variable: A dictionay in which
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: s/dictionay/dictionary/

File: chainer/links/model/vision/vgg.py [hunk 4]
```diff
@@ -188,8 +185,9 @@ def forward(self, x, layers=None, **kwargs):
 
         if kwargs:
             argument.check_unexpected_kwargs(
-                kwargs, test='test argument is not supported anymore. '
-                'Use chainer.using_config')
+                kwargs, test='test argument is not supported anymote. '
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: This change seems unneeded (and typo: `anymote`)

File: chainer/links/model/vision/vgg.py [hunk 5]
```diff
@@ -210,7 +208,7 @@ def extract(self, images, layers=None, size=(224, 224), **kwargs):
 
         Extracts all the feature maps of given images.
 
-        The difference of directly executing ``forward`` is that
+        The difference of directry executing ``forward`` is that
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: This change seems unneeded.

File: chainer/links/model/vision/vgg.py [hunk 6]
```diff
@@ -31,17 +31,17 @@
 from chainer.variable import Variable
 
 
-class VGG16Layers(link.Chain):
+class VGGLayers(link.Chain):
 
-    """A pre-trained CNN model with 16 layers provided by VGG team.
+    """A pre-trained CNN model provided by VGG team.
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: Please add:
```
You can use VGG16Layers or VGG19Layers for concrete implementations.
```

File: chainer/links/model/vision/vgg.py (hunk 1 above)
Comment (crcrpar) [reply chainer/links/model/vision/vgg.py]: So, This was because I did not want to define the default value for `n_layers` and now I know I should change the order of arguments.

File: chainer/links/model/vision/vgg.py (hunk 2 above)
Comment (crcrpar) [reply chainer/links/model/vision/vgg.py]: OK

File: chainer/links/model/vision/vgg.py (hunk 6 above)
Comment (crcrpar) [reply chainer/links/model/vision/vgg.py]: Thanks

File: chainer/links/model/vision/vgg.py (hunk 1 above)
Comment (crcrpar) [reply chainer/links/model/vision/vgg.py]: Ah, I missed the ResNet imple. I will follow the style.

File: chainer/links/model/vision/vgg.py [hunk 7]
```diff
@@ -66,26 +68,33 @@ class VGG16Layers(link.Chain):
...
+    def __init__(self, pretrained_model='auto', n_layers=16):
+        super(VGGLayers, self).__init__()
         if pretrained_model:
-            # As a sampling process is time-consuming,
+            # Because a sampling process is time-consuming,
             # we employ a zero initializer for faster computation.
             init = constant.Zero()
             kwargs = {'initialW': init, 'initial_bias': init}
         else:
-            # employ default initializers used in the original paper
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: Why is this comment removed?

File: chainer/links/model/vision/vgg.py [hunk 8]
```diff
@@ -66,26 +68,33 @@ class VGG16Layers(link.Chain):
...
             used by ``forward`` and ``extract`` methods.
 
     """
 
-    def __init__(self, pretrained_model='auto'):
+    def __init__(self, pretrained_model='auto', n_layers=16):
+        super(VGGLayers, self).__init__()
         if pretrained_model:
-            # As a sampling process is time-consuming,
+            # Because a sampling process is time-consuming,
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: Why is this changed? (the original sentence seems OK)

File: chainer/links/model/vision/vgg.py [hunk 9]
```diff
@@ -66,26 +68,33 @@ class VGG16Layers(link.Chain):
...
 
     Attributes:
         available_layers (list of str): The list of available layer names
             used by ``forward`` and ``extract`` methods.
 
     """
 
-    def __init__(self, pretrained_model='auto'):
+    def __init__(self, pretrained_model='auto', n_layers=16):
+        super(VGGLayers, self).__init__()
```
Comment (kmaehashi) [inline chainer/links/model/vision/vgg.py]: Why is this code moved?

Comment (crcrpar) [reply chainer/links/model/vision/vgg.py]: Because I added the comment "You can use ... VGG16Layers and VGG19Layers" at the beginning of the above docstring.

Comment (crcrpar) [reply chainer/links/model/vision/vgg.py]: If i mistakenly put the explanation in wrong place, please tell me the exact/appropriate line. +)

File: chainer/links/model/vision/vgg.py (hunk 7 above)
Comment (crcrpar) [reply chainer/links/model/vision/vgg.py]: sorry to miss this.

File: chainer/links/model/vision/vgg.py (hunk 9 above)
Comment (kmaehashi) [reply chainer/links/model/vision/vgg.py]: Ok, I initially meant to ask why the line `super(VGGLayers, self).__init__()` has been moved, but the current code is also fine.

**Comment by kmaehashi** [comment]
LGTM except issues about comments (as commented above). Could you update the PR?

**Comment by crcrpar** [comment]
ok. I reflected the comments.

Am I responsible for this failure?

**Comment by kmaehashi** [comment]
Thanks! The test failures seems not related to this PR.

Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM! Thank you!
